### PR TITLE
[IOTDB-1207]Open time partition causes stackoverflow

### PR DESCRIPTION
**Pull request by yanhongwangg**
https://issues.apache.org/jira/browse/IOTDB-1207

### CODE REVIEW

**Review by mychaow — APPROVED** [review]
LGTM

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java [hunk 1]
```diff
@@ -321,17 +321,11 @@ public TimeValuePair getLastCache(PartialPath seriesPath) {
...
-        plan.setMeasurementMNodes(measurementMNodes);
-        return deviceNode;
-      }
-    } catch (PathNotExistException exception) {
-      // ignore, so we could try local MTree
+    int nonExistSchemaIndex =
+        getMNodesLocally(plan.getDeviceId(), plan.getMeasurements(), measurementMNodes);
+    if (nonExistSchemaIndex == -1) {
+      plan.setMeasurementMNodes(measurementMNodes);
+      return new MNode(null, plan.getDeviceId().getDevice());
```
Comment (wangchao316) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:328]: new MNode, which could add to cache?   Does the memory overhead increase if new is used each time?

**Review by neuyilan — APPROVED** [review]
LGTM

**Review by LebronAl — APPROVED** [review]
LGTM.
Looking forward to your solution to the deadlock problem described in this issue in next PR.

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 1 above)
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:328]: In cluster module, the return value of this function is not cached and is only used to detect whether an exception will be thrown.
In addition, an MNode instance is not very memory intensive, and the variables are null by default, so the overhead in memory should not be very large

**Pull request merged**